### ISSUE TITLE
Dagger 2.40 issue when syncing gradle

### ISSUE THREAD

**Issue by AlfredAndroidTedmob** [google/dagger#2990]
After updating from version 2.39.1 to version 2.40, the build failure occurs when syncing gradle (error logs below).
[dagger-2.40-syncing-issue.txt](https://github.com/google/dagger/files/7472524/dagger-2.40-syncing-issue.txt)

I think the problem is:
`org.gradle.api.UnknownDomainObjectException: KotlinJvmAndroidCompilation with name 'prodReleaseUnitTest' not found`

I could not find any satisfactory explanations on how to deal with this error.

**Some details about the project**:
- gradle version 7.0.3 (Android Studio 2020.3.1 Patch 3)
- buildTypes: debug and release.

```
flavorDimensions "scope"

productFlavors {
    prod {
        dimension "scope"
    }
}
```
- There are no test folders in the project (also, the test dependencies were removed).
- The same error occurs when syncing using prodDebug or prodRelease.

**Working scenarios**:
1- Using version 2.39.1 or below
2- Adding this to the build.gradle file in the android config:
`hilt {
        enableAggregatingTask = true
    }`

Any help is greatly appreciated, as we have to update around 10 to 15 android projects, and we prefer using Dagger's recommendation (of keeping `enableAggregatingTask` to true).

**Comment by danysantiago** [comment]
Can you share more information about your project?

What other plugins are being applied to it? We've seen other plugins conflict with Hilt: https://github.com/google/dagger/issues/2744#issuecomment-901277926

Any custom task configuration or dependency resolution?

**Comment by AlfredAndroidTedmob** [comment]
> 
> 
> Can you share more information about your project?
> 
> What other plugins are being applied to it? We've seen other plugins conflict with Hilt: [#2744 (comment)](https://github.com/google/dagger/issues/2744#issuecomment-901277926)
> 
> Any custom task configuration or dependency resolution?

Those are the plugins we use in our projects (in the order shown, not sure if it matters for Dagger, but it does for FirebaseCrashlytics):
```
apply plugin: 'com.onesignal.androidsdk.onesignal-gradle-plugin'
apply plugin: 'com.android.application'
apply plugin: 'kotlin-android'
apply plugin: 'kotlin-kapt'
apply plugin: 'dagger.hilt.android.plugin'
apply plugin: 'com.google.gms.google-services'
apply plugin: 'com.google.firebase.crashlytics'
apply plugin: 'androidx.navigation.safeargs.kotlin'
apply plugin: 'kotlin-parcelize'
```

As far as tasks, we do not have our own, except the delete build file on clean:
```
tasks.register("delete", Delete) {
    delete rootProject.buildDir
}
```

We don't have custom dependency resolution, but I will have to try your workaround (**since it seems it may be related to Google Play Services, as mentioned in** [#2744 (comment)](https://github.com/google/dagger/issues/2744#issuecomment-901277926)).
We currently have the below dependency:
`implementation platform("com.google.firebase:firebase-bom:29.0.0")`

**Comment by AlfredAndroidTedmob** [comment]
> 
> 
> > Can you share more information about your project?
> > What other plugins are being applied to it? We've seen other plugins conflict with Hilt: [#2744 (comment)](https://github.com/google/dagger/issues/2744#issuecomment-901277926)
> > Any custom task configuration or dependency resolution?
> 
> Those are the plugins we use in our projects (in the order shown):
> 
> ```
> apply plugin: 'com.onesignal.androidsdk.onesignal-gradle-plugin'
> apply plugin: 'com.android.application'
> apply plugin: 'kotlin-android'
> apply plugin: 'kotlin-kapt'
> apply plugin: 'dagger.hilt.android.plugin'
> apply plugin: 'com.google.gms.google-services'
> apply plugin: 'com.google.firebase.crashlytics'
> apply plugin: 'androidx.navigation.safeargs.kotlin'
> apply plugin: 'kotlin-parcelize'
> ```
> 
> As far as tasks, we do not have our own, except the delete build file on clean:
> 
> ```
> tasks.register("delete", Delete) {
>     delete rootProject.buildDir
> }
> ```
> 
> We don't have custom dependency resolution, but I will have to try your workaround (**since it seems it may be related to Google Play Services, as mentioned in** [#2744 (comment)](https://github.com/google/dagger/issues/2744#issuecomment-901277926)). We currently have the below dependency: `classpath 'com.google.gms:google-services:4.3.10'`

It seems the solution from [#2744 (comment)](https://github.com/google/dagger/issues/2744#issuecomment-901277926) will not work in my project, since I cannot find the "getDependencies" task using: `gradlew task --all`.

**Comment by danysantiago** [comment]
The solution in https://github.com/google/dagger/issues/2744#issuecomment-901277926 is if you have the `oss-licenses-plugin` applied but it seems you don't. It would be nice if you could drill down which of the plugins is a conflict by commenting it out temporarily.

Maybe its `com.onesignal.androidsdk.onesignal-gradle-plugin`, `com.google.gms.google-services` or `com.google.firebase.crashlytics`.

Meanwhile we are investigating this as we've also seen a conflict with AGP's `CheckJetifier` task.

**Comment by AlfredAndroidTedmob** [comment]
> 
> 
> The solution in [#2744 (comment)](https://github.com/google/dagger/issues/2744#issuecomment-901277926) is if you have the `oss-licenses-plugin` applied but it seems you don't. It would be nice if you could drill down which of the plugins is a conflict by commenting it out temporarily.
> 
> Maybe its `com.onesignal.androidsdk.onesignal-gradle-plugin`, `com.google.gms.google-services` or `com.google.firebase.crashlytics`?
> 
> Meanwhile we are investigating this as we've also seen a conflict with AGP's `CheckJetifier` task.

Yes, it does seem like it is related to OneSignal.
More specifically, the gradle dependency conflict resolver plugin: "gradle.plugin.com.onesignal:onesignal-gradle-plugin".
I removed the plugin, and started adding the "play-services-base" needed, and the project synced **successfully**.

I will have to make sure since removing the dependency resolver means I have to change a lot of the Google Services dependencies (I will have to rely on the same version used bt the Firebase BoM to make sure Crashlytics is not the source of the problem).

I have not found issues in the [OneSignal](https://github.com/OneSignal/OneSignal-Android-SDK) project related to this as of now. Should I raise an issue pointing to this or any other issue to OneSignal in case I can confirm this?
I don't have much time now due to my job, but I will try.

**Comment by danysantiago** [comment]
Yes, please raise an issue in the [OneSignal Gradle Plugin project](https://github.com/OneSignal/OneSignal-Gradle-Plugin).
